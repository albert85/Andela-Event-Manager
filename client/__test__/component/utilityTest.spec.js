import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import CenterTableRow from '../../src/utility/centerTableRow';

let isComponentMounted;

configure({ adapter: new Adapter() });


describe('Index Routes', () => {
  const props = {
    centers: {
      name: '',
      location: '',
      amount: '',
      capacity: '',
    },
    handleCenterDetails: () => {},
    handleEditCenterDetails: () => {},
  };

  const getComponentStatus = () => {
    if (!isComponentMounted) {
      isComponentMounted = shallow(<CenterTableRow {...props} />);
    }
    return isComponentMounted;
  };

  it('should successfully rendered', () => {
    expect(getComponentStatus()).toMatchSnapshot();
  });

  it('should successfully rendered', () => {
    expect(getComponentStatus()).toHaveLength(1);
  });
});
