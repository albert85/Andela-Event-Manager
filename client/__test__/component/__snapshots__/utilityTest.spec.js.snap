// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Routes should successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CenterRows
    centers={
      Object {
        "amount": "",
        "capacity": "",
        "location": "",
        "name": "",
      }
    }
    handleCenterDetails={[Function]}
    handleEditCenterDetails={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td
          scope="row"
        >
          NaN
        </td>,
        <td>
          
        </td>,
        <td>
          
        </td>,
        <td>
          
        </td>,
        <td>
          
        </td>,
        <td>
          <div
            className="row"
          >
            <div
              className="col mb-2"
            >
              <button
                className="btn btn-primary btn-block"
                onClick={[Function]}
                type="button"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-pencil"
                >
                   Edit
                </i>
              </button>
            </div>
            <div
              className="col"
            >
              <button
                className="btn btn-success btn-block"
                onClick={[Function]}
                type="button"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-book"
                >
                   Update
                </i>
              </button>
            </div>
          </div>
        </td>,
      ],
      "className": "border border-white",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": NaN,
          "scope": "row",
        },
        "ref": null,
        "rendered": NaN,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="col mb-2"
            >
              <button
                className="btn btn-primary btn-block"
                onClick={[Function]}
                type="button"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-pencil"
                >
                   Edit
                </i>
              </button>
            </div>
            <div
              className="col"
            >
              <button
                className="btn btn-success btn-block"
                onClick={[Function]}
                type="button"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-book"
                >
                   Update
                </i>
              </button>
            </div>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col mb-2"
              >
                <button
                  className="btn btn-primary btn-block"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-pencil"
                  >
                     Edit
                  </i>
                </button>
              </div>,
              <div
                className="col"
              >
                <button
                  className="btn btn-success btn-block"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-book"
                  >
                     Update
                  </i>
                </button>
              </div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn btn-primary btn-block"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-pencil"
                  >
                     Edit
                  </i>
                </button>,
                "className": "col mb-2",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    aria-hidden="true"
                    className="fa fa-pencil"
                  >
                     Edit
                  </i>,
                  "className": "btn btn-primary btn-block",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "children": " Edit",
                    "className": "fa fa-pencil",
                  },
                  "ref": null,
                  "rendered": " Edit",
                  "type": "i",
                },
                "type": "button",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn btn-success btn-block"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-book"
                  >
                     Update
                  </i>
                </button>,
                "className": "col",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    aria-hidden="true"
                    className="fa fa-book"
                  >
                     Update
                  </i>,
                  "className": "btn btn-success btn-block",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "children": " Update",
                    "className": "fa fa-book",
                  },
                  "ref": null,
                  "rendered": " Update",
                  "type": "i",
                },
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td
            scope="row"
          >
            NaN
          </td>,
          <td>
            
          </td>,
          <td>
            
          </td>,
          <td>
            
          </td>,
          <td>
            
          </td>,
          <td>
            <div
              className="row"
            >
              <div
                className="col mb-2"
              >
                <button
                  className="btn btn-primary btn-block"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-pencil"
                  >
                     Edit
                  </i>
                </button>
              </div>
              <div
                className="col"
              >
                <button
                  className="btn btn-success btn-block"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-book"
                  >
                     Update
                  </i>
                </button>
              </div>
            </div>
          </td>,
        ],
        "className": "border border-white",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": NaN,
            "scope": "row",
          },
          "ref": null,
          "rendered": NaN,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
            >
              <div
                className="col mb-2"
              >
                <button
                  className="btn btn-primary btn-block"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-pencil"
                  >
                     Edit
                  </i>
                </button>
              </div>
              <div
                className="col"
              >
                <button
                  className="btn btn-success btn-block"
                  onClick={[Function]}
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-book"
                  >
                     Update
                  </i>
                </button>
              </div>
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col mb-2"
                >
                  <button
                    className="btn btn-primary btn-block"
                    onClick={[Function]}
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-pencil"
                    >
                       Edit
                    </i>
                  </button>
                </div>,
                <div
                  className="col"
                >
                  <button
                    className="btn btn-success btn-block"
                    onClick={[Function]}
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-book"
                    >
                       Update
                    </i>
                  </button>
                </div>,
              ],
              "className": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="btn btn-primary btn-block"
                    onClick={[Function]}
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-pencil"
                    >
                       Edit
                    </i>
                  </button>,
                  "className": "col mb-2",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      aria-hidden="true"
                      className="fa fa-pencil"
                    >
                       Edit
                    </i>,
                    "className": "btn btn-primary btn-block",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "children": " Edit",
                      "className": "fa fa-pencil",
                    },
                    "ref": null,
                    "rendered": " Edit",
                    "type": "i",
                  },
                  "type": "button",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="btn btn-success btn-block"
                    onClick={[Function]}
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-book"
                    >
                       Update
                    </i>
                  </button>,
                  "className": "col",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      aria-hidden="true"
                      className="fa fa-book"
                    >
                       Update
                    </i>,
                    "className": "btn btn-success btn-block",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "children": " Update",
                      "className": "fa fa-book",
                    },
                    "ref": null,
                    "rendered": " Update",
                    "type": "i",
                  },
                  "type": "button",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
